File: app/src/main/java/net/tlalka/android/fiszki/view/activities/TestScoreActivity.java
package net.tlalka.android.fiszki.view.activities;

import javax.inject.Inject;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.OnClick;
import net.tlalka.android.fiszki.R;
import net.tlalka.android.fiszki.model.dto.LessonDto;
import net.tlalka.android.fiszki.view.navigations.Navigator;

public class TestScoreActivity extends BasePageActivity {

    @BindView(R.id.test_topic)
    protected TextView testTopic;

    @BindView(R.id.test_score_total)
    protected Button testTotal;

    @BindView(R.id.test_score_value)
    protected TextView testScore;

    @BindView(R.id.test_score_incorrect)
    protected TextView testIncorrect;

    @Inject
    protected Navigator navigator;

    @Inject
    protected LessonDto lessonDto;

    @Override
    public void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        super.setContentView(R.layout.test_score_activity);
        super.getActivityComponent().inject(this);

        this.runActivity();
    }

    private void runActivity() {
        int lessonIndex = this.lessonDto.getLessonIndex();
        String lessonName = this.lessonDto.getLessonName();

        this.testTopic.setText(getString(R.string.test_topic, lessonIndex, lessonName));
        this.testTotal.setText(getString(R.string.test_score_total, lessonDto.getCorrectScore()));
        this.testScore.setText(getString(R.string.test_score_value, lessonDto.getGeneralScore()));
        this.testIncorrect.setText(getString(R.string.test_score_incorrect, lessonDto.getIncorrectScore()));
    }

    @OnClick(R.id.test_score_repeat)
    public void onRepeatClick(View view) {
        this.navigator.openTestActivity(this, this.lessonDto);
        this.navigator.finish(this);
    }

    @OnClick(R.id.test_score_tests)
    public void onTestsClick(View view) {
        this.navigator.openTestListActivity(this);
        this.navigator.finish(this);
    }
}

File: app/src/main/java/net/tlalka/android/fiszki/view/activities/TestScoreActivity.kt
package net.tlalka.android.fiszki.view.activities

import javax.inject.Inject

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView

import butterknife.BindView
import butterknife.OnClick
import net.tlalka.android.fiszki.R
import net.tlalka.android.fiszki.model.dto.LessonDto
import net.tlalka.android.fiszki.view.navigations.Navigator

class TestScoreActivity : BasePageActivity() {

    @BindView(R.id.test_topic)
    lateinit var testTopic: TextView

    @BindView(R.id.test_score_total)
    lateinit var testTotal: Button

    @BindView(R.id.test_score_value)
    lateinit var testScore: TextView

    @BindView(R.id.test_score_incorrect)
    lateinit var testIncorrect: TextView

    @Inject
    lateinit var navigator: Navigator

    @Inject
    lateinit var lessonDto: LessonDto

    public override fun onCreate(bundle: Bundle?) {
        super.onCreate(bundle)
        super.setContentView(R.layout.test_score_activity)
        super.activityComponent.inject(this)

        runActivity()
    }

    private fun runActivity() {
        testTopic.text = getString(R.string.test_topic, lessonDto.lessonIndex, lessonDto.lessonName)
        testTotal.text = getString(R.string.test_score_total, lessonDto.correctScore)
        testScore.text = getString(R.string.test_score_value, lessonDto.generalScore)
        testIncorrect.text = getString(R.string.test_score_incorrect, lessonDto.incorrectScore)
    }

    @OnClick(R.id.test_score_repeat)
    fun onRepeatClick(view: View) {
        navigator.openTestActivity(this, this.lessonDto)
        navigator.finish(this)
    }

    @OnClick(R.id.test_score_tests)
    fun onTestsClick(view: View) {
        navigator.openTestListActivity(this)
        navigator.finish(this)
    }
}
